controllers/binding/snapshotenvironmentbinding_adapter_test.go

```diff
@@ -505,14 +458,25 @@ var _ = Describe("Binding Adapter", Ordered, func() {
 				ContextKey: loader.EnvironmentContextKey,
 				Resource:   hasEnv,
 			},
+			{
+				ContextKey: loader.SnapshotContextKey,
+				Resource:   hasSnapshot,
+			},
 			{
 				ContextKey: loader.DeploymentTargetContextKey,
 				Resource:   deploymentTarget,
 			},
+			{
+				ContextKey: loader.DeploymentTargetClaimContextKey,
+				Resource:   deploymentTargetClaim,
+			},
 		})
 
-		dtc, _ := adapter.loader.GetDeploymentTargetClaimForEnvironment(k8sClient, adapter.context, hasEnv)
-		Expect(dtc).NotTo(BeNil())
+		Eventually(func() error {
```

Comment:
We don't need this `Eventually` as long as we use mocked dtc, the mocked dtc won't be really deleted.

Reply:
Then I don't understand why the error in STONEINTG-595 happened. Can you please explain it ?

Reply:
> We don't need this `Eventually` as long as we use mocked dtc, the mocked dtc won't be really deleted.

Here is waiting the DTC created in `BeforeEach`.

Reply:
Change to use DTC from mockedData.

Reply:
@hongweiliu17, please help to review it again.